src/Bicep.Core.IntegrationTests/ScenarioTests.cs

```diff
@@ -1640,5 +1640,38 @@ public void Test_Issue2268()
             evaluated.Should().HaveValueAtPath("$.resources[0].name", "myServer/myDb/myTde");
             evaluated.Should().HaveValueAtPath("$.outputs['tdeId'].value", "/subscriptions/f91a30fd-f403-4999-ae9f-ec37a6d81e13/resourceGroups/testResourceGroup/providers/Microsoft.Sql/servers/myServer/databases/myDb/transparentDataEncryption/myTde");
         }
+
+        [TestMethod]
+        // https://github.com/azure/bicep/issues/1809
+        public void Test_Issue1809()
+        {
+            var result = CompilationHelper.Compile(
+                ("main.bicep", @"
+module tags './tags.bicep' = {
+  name: 'tags'
+}
+
```

Comment:
Mind also adding specific test(s) for for-expressions just to make sure there's no feature interaction there?

Reply:
Good pt. Added tests for for-expressions.